src/audio/copier/copier.c

```diff
@@ -1653,10 +1653,16 @@ static uint64_t copier_get_processed_data(struct comp_dev *dev, uint32_t stream_
 {
 	struct copier_data *cd = comp_get_drvdata(dev);
 	uint64_t ret = 0;
+	bool source = dev->direction == SOF_IPC_STREAM_CAPTURE;
 
 	if (cd->endpoint_num && cd->bsource_buffer != input) {
-		if (stream_no < cd->endpoint_num)
-			ret = comp_get_total_data_processed(cd->endpoint[stream_no], 0, input);
+		if (stream_no < cd->endpoint_num) {
+			if (dev->ipc_config.type == SOF_COMP_HOST)
+				ret = host_zephyr_get_processed_data(cd->hd, 0, input, source);
+			else
+				ret = comp_get_total_data_processed(cd->endpoint[stream_no],
+								    0, input);
+		}
```

Comment:
Seeing lots of
```
if (host)
    zephyr_call()
else
    other_call()
```
Is the else condition legacy ?

Reply:
@lgirdwood this PR is one half of the final optimization effort as it covers only host copier parts. The else is for this DAI copier here and everywhere else. The DAI copier will be optimized in a follow up PR.

Reply:
That's correct, I also saw this part is not beautiful and feel not good, however, once DAI one more time copy removed in future, there should be no if / else, then we can get beautiful version code for copier, look forward to see that version after 1 month, :).